core/src/parking_lot.rs

```diff
@@ -177,7 +179,25 @@ impl ThreadData {
         // Keep track of the total number of live ThreadData objects and resize
         // the hash table accordingly.
         let num_threads = NUM_THREADS.fetch_add(1, Ordering::Relaxed) + 1;
-        grow_hashtable(num_threads);
+
+        // `grow_hashtable` below performs global allocation. If the global
+        // allocator uses parking_lot locks, it'll attempt to acquire a lock,
+        // which will re-enter this `new` function. When the
+        // "global_allocator_compat" feature is enabled, detect this case
+        // and skip growing the hash table until the inner call returns,
+        // letting the outer call perform the grow.
+        #[cfg(feature = "global_allocator_compat")]
+        let grow = {
+            thread_local!(static ACTIVE: Cell<bool> = Cell::new(false));
```

Comment:
This seems unreliable, if the target platform uses pthread tls, there will be multiple memory operations inside std.

see https://github.com/rust-lang/rust/blob/master/library/std/src/thread/local.rs#L740 https://github.com/rust-lang/rust/blob/master/library/std/src/sys_common/thread_local_dtor.rs#L19